kratos/processes/skin_detection_process.cpp

```diff
@@ -0,0 +1,217 @@
+            }
+        }
+    }
+
+    // Create the face_set
+    HashSetVectorIntType face_set;
+
+    for(IndexType i = 0; i < number_of_elements; ++i) {
+        auto it_elem = mrModelPart.Elements().begin() + i;
+
+        GeometryType& geom = it_elem->GetGeometry();
+
+        const SizeType potential_number_neighbours = ComputePotentialNeighboursSize(it_elem);
+
+        for (IndexType i_face = 0; i_face < potential_number_neighbours; ++i_face) {
+
+            /* FACES/EDGES */
+            const SizeType number_nodes = TDim == 2 ? geom.Edges()[i_face].size() : geom.Faces()[i_face].size();
+            VectorIndexType vector_ids(number_nodes);
+
+            /* FACE/EDGE */
+            if (TDim == 2) {
+                for (IndexType i_node = 0; i_node < number_nodes; ++i_node) {
+                    vector_ids[i_node] = geom.Edges()[i_face][i_node].Id();
+                }
+            } else {
+                for (IndexType i_node = 0; i_node < number_nodes; ++i_node) {
+                    vector_ids[i_node] = geom.Faces()[i_face][i_node].Id();
+                }
+            }
+
+            /*** THE ARRAY OF IDS MUST BE ORDERED!!! ***/
+            std::sort(vector_ids.begin(), vector_ids.end());
+            // Check if the elements already exist in the HashSetVectorIntType
+            HashSetVectorIntTypeIteratorType it_check = face_set.find(vector_ids);
+
+            if(it_check != face_set.end() ) {
+                // If it exists we remove from the inverse map
+                inverse_face_map.erase(vector_ids);
+            } else {
+                // If it doesn't exist it is added to the database
+                face_set.insert(vector_ids);
+            }
+        }
+    }
+
+    // We create the auxiliar ModelPart
+    ModelPart::Pointer p_auxiliar_model_part = mrModelPart.CreateSubModelPart(mThisParameters["name_auxiliar_model_part"].GetString());
+
+    // The auxiliar name of the condition
+    const std::string& name_condition = mThisParameters["name_auxiliar_condition"].GetString();
+    const std::string pre_name = TDim == 2 ? "" : "Surface";
+
+    // The number of conditions
+    IndexType condition_id = mrModelPart.Conditions().size();
+
+    // The indexes of the nodes of the skin
+    std::unordered_set<IndexType> nodes_in_the_skin;
+
+    // Create the auxiliar conditions
+    Properties::Pointer p_prop_0 = mrModelPart.pGetProperties(0);
+    for (auto& map : inverse_face_map) {
+        condition_id += 1;
+
+        const VectorIndexType& nodes_face = map.second;
+
+        for (auto& index : nodes_face)
+            nodes_in_the_skin.insert(index);
+
+        const std::string complete_name = pre_name + name_condition + std::to_string(TDim) + "D" + std::to_string(nodes_face.size()) + "N"; // If the condition doesn't follow this structure...sorry, we then need to modify this...
+        if (TDim == 2) {
```

Comment:
I think the if is not needed

Reply:
After the last change I did, it is not, thanks